sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -102,9 +104,63 @@ export function convertResponseForQueryBatch(
       }))
   };
 
+  (newResponse as any)["__fixApplied"] = fixApplied;
   return newResponse;
 }
 
+/**
+ * This is a workaround for a service bug that we're investigating. The 'body' column will occasionally come
+ * back as a JSON string, instead of being a JSON object.
+ *
+ * (examples, with excess stuff trimmed)
+ * Correct: `{"responses":[{"body":{"tables":[{"name":"PrimaryResult","columns":[{"name":"stringcolumn","type":"string"}],"rows":[["hello"]}`
+ * Broken: `{"responses":[{"body":"{\"tables\":[{\"name\":\"PrimaryResult\",\"columns\":[{\"name\":\"stringcolumn\",\"type\":\"string\"}],\"rows\":[[\"hello\"]}`
+ *
+ * Issue here: https://github.com/Azure/azure-sdk-for-js/issues/15688
+ *
+ * @internal
+ */
+export function fixInvalidBatchQueryResponse(
+  generatedResponse: GeneratedQueryBatchResponse
+): boolean {
+  if (generatedResponse.responses == null) {
+    return false;
+  }
+
+  let wholeResponse: GeneratedQueryBatchResponse | undefined;
+  let hadToFix = false;
+
+  // fix whichever responses are in this broken state (each query has it's own
+  // response, so they're not all always broken)
+  for (let i = 0; i < generatedResponse.responses.length; ++i) {
+    if (
+      generatedResponse.responses[i].body?.tables != null ||
+      generatedResponse.responses[i].body?.error != null
+    ) {
+      continue;
+    }
+
+    // the body here is incorrect, deserialize the correct one from the raw response itself.
+
+    // deserialize the raw response from the service, since we'll need index into it.
+    if (!wholeResponse) {
+      wholeResponse = JSON.parse(
+        generatedResponse["_response"].bodyAsText
```

Comment:
If this is using core-v2 maybe we should use `onResponse` callback instead?

Reply:
Not yet using core-v2 but I think you're right (from my understanding of how it was rearchitected).